From ae71375ad5ebeef5c06189e4451ddaf9b7139cb8 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Mon, 6 Feb 2023 10:14:19 +0800
Subject: [PATCH 09/45] Revert "drm/msm/adreno: Make adreno quirks not
 overwrite each other"

This reverts commit 8c71777b6a17527ff0bb12a973193f481fa7fc5e.
---
 drivers/gpu/drm/msm/adreno/adreno_gpu.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/msm/adreno/adreno_gpu.h b/drivers/gpu/drm/msm/adreno/adreno_gpu.h
index 4656e707fe27d..c3775f79525a7 100644
--- a/drivers/gpu/drm/msm/adreno/adreno_gpu.h
+++ b/drivers/gpu/drm/msm/adreno/adreno_gpu.h
@@ -28,9 +28,11 @@ enum {
 	ADRENO_FW_MAX,
 };
 
-#define ADRENO_QUIRK_TWO_PASS_USE_WFI		BIT(0)
-#define ADRENO_QUIRK_FAULT_DETECT_MASK		BIT(1)
-#define ADRENO_QUIRK_LMLOADKILL_DISABLE		BIT(2)
+enum adreno_quirks {
+	ADRENO_QUIRK_TWO_PASS_USE_WFI = 1,
+	ADRENO_QUIRK_FAULT_DETECT_MASK = 2,
+	ADRENO_QUIRK_LMLOADKILL_DISABLE = 3,
+};
 
 struct adreno_rev {
 	uint8_t  core;
@@ -60,7 +62,7 @@ struct adreno_info {
 	const char *name;
 	const char *fw[ADRENO_FW_MAX];
 	uint32_t gmem;
-	u64 quirks;
+	enum adreno_quirks quirks;
 	struct msm_gpu *(*init)(struct drm_device *dev);
 	const char *zapfw;
 	u32 inactive_period;
-- 
2.25.1

